source/elements/oneMKL/source/domains/value_or_pointer.rst

```diff
@@ -0,0 +1,106 @@
+.. SPDX-FileCopyrightText: 2019-2020 Intel Corporation
+..
+.. SPDX-License-Identifier: CC-BY-4.0
+
+.. _value_or_pointer:
+
+Scalar Arguments in BLAS
+========================
+
+.. container::
+
+   The USM version of oneMKL BLAS routines for DPC++ will accept either
+   a scalar (for example ``float``) or pointer (``float*``) for parameters
+   that represent a single fixed value (not a vector or matrix). These
+   parameters are often named ``alpha`` or ``beta`` in BLAS.
+
+   .. container:: section
+
+      .. rubric:: Basic Use
+         :name: basic-use
+         :class: sectiontitle
+
+      Users can call ``gemv`` with pointers:
+
+        .. code-block::
+
+          float *alpha_ptr = sycl::malloc_shared<float>(1, queue);
+          float *beta_ptr = sycl::malloc_shared<float>(1, queue);
+          // fill alpha_ptr and beta_ptr with desired values
+          oneapi::mkl::blas::column_major::gemv(queue, trans, m, n, alpha_ptr, lda, x, incx, beta_ptr,
+                                                y, incy).wait();
+
+      or with literal values:
+
+        .. code-block::
+
+          oneapi::mkl::blas::column_major::gemv(queue, trans, m, n, 2, lda, x, incx, 2.7,
+                                                y, incy).wait();
+
+      Users can even mix scalar and pointer parameters in a single call:
+
+        .. code-block::
+
+          float *alpha_ptr = sycl::malloc_shared<float>(1, queue);
+          oneapi::mkl::blas::column_major::gemv(queue, trans, m, n, alpha_ptr, lda, x, incx, 2.7,
+                                             y, incy).wait();
+
+      Pointers provided for scalar parameters may be SYCL-managed pointers
+      to either device or host memory (for example pointers created with
+      ``sycl::malloc_device``, ``sycl::malloc_shared``, or
+      ``sycl::malloc_host``), or they may be raw pointers created with
+      ``malloc`` or ``new``.
+
+      For most users, this is all they need to know. A few details about how
+      this is implemented are provided below.
+
+   .. container:: section
+
+      .. rubric:: Wrapper type
+         :name: wrapper-time
+         :class: sectiontitle
+
+      The USM version of oneMKL BLAS routines use a templated
+      ``value_or_pointer<T>`` wrapper to enable either pointers or values
+      to be passed to routines that take a scalar parameter. This type is in
+      the ``oneapi::mkl::`` namespace and defined in the header file
+      ``oneapi/mkl/types.hpp``.
```

Comment:
This seems to be too specific to oneMKL product. For spec, should we consider removing this or make it more general?

Reply:
Agreed, I've removed that sentence.